src/statemachine.rs

```diff
@@ -232,8 +254,12 @@ impl StateMachine {
                     PinUvAuthResult::SuccessGetPinUvAuthTokenUsingUvWithPermissions,
                     pin_auth_token,
                 )
-            } else if supports_pin && pin_configured {
+            } else {
                 // CTAP 2.1 - PIN
+                // We would have exited earlier if !pin_configured && !supports_uv,
+                // So we have pin_configured || (skip_uv && supports_uv)
+                // TODO(https://github.com/mozilla/authenticator-rs/issues/223)
+                // It should not be possible to arrive here in the !pin_configured case.
```

Comment:
I don't think this comment is true? 
Suppose, there is no PIN configured, but the token supports UV. And we skip UV. Then we will end up here.
The early exit does not take `skip_uv` into account.

Reply:
Yes, you're right.